app/models/push_notification_target.rb

```diff
@@ -0,0 +1,113 @@
+class PushNotificationTarget < ApplicationRecord
+  resourcify
+
+  FOREM_BUNDLE = "com.forem.app".freeze
+  SUPPORTED_PLATFORMS = [Device::ANDROID, Device::IOS].freeze
+  FOREM_APP_PLATFORMS = [Device::IOS].freeze
+
+  validates :app_bundle, presence: true
+  validates :platform, presence: true
+
+  # Clear Redis-backed model to ensure it will be recreated with updated values
+  after_update :clear_rpush_app
+
+  def forem_app?
+    app_bundle == FOREM_BUNDLE
+  end
+
+  def active?
+    # TRUE if it's marked as `active` in the DB && it has credentials available
+    active && auth_credentials?
+  end
+
+  def auth_credentials?
+    auth_credentials.present?
+  end
+
+  def auth_credentials
+    if app_bundle == FOREM_BUNDLE && platform == Device::IOS
+      ApplicationConfig["RPUSH_IOS_PEM"]
+    else
+      auth_key
+    end
+  end
+
+  def recreate_ios_app!
+    app = Rpush::Apns2::App.new
+    app.name = app_bundle
+    app.certificate = auth_credentials.to_s.gsub("\\n", "\n")
+    app.environment = Rails.env.production? ? "production" : "development"
```

Comment:
```ruby
[10] pry(main)> Rails.env
=> "development"
```

I think you can just do `app.environment = Rails.env`

Reply:
That would definitely clean this up! 

I'm a bit worried about the test environment though. I'm not sure if `app.environment = "test"` would break (Rpush model validations), but if it works I'll keep this suggestion 🙌

Reply:
Update: It worked 🔥